Fix and improve Gemini CLI troubleshooting.md doc (#2734)
google-gemini/gemini-cli: docs/troubleshooting.md

# Troubleshooting guide

> This guide provides solutions to common issues and debugging tips, including topics on:

Author: or 
```suggestion
This guide provides answers and solutions to common issues, as well as debugging tips, covering:
```

Author: With a small table of contents it may be easier to follow

    associated with their Gmail accounts may not be able to activate the free
    tier of the Google Code Assist plan.
    - For Google Cloud accounts, you can work around this by setting
    `GOOGLE_CLOUD_PROJECT` to your project ID.
    - Alternatively, you can also grab an API key from [AI
    Studio](http://aistudio.google.com/app/apikey), which also includes a
    separate free tier.

## Frequently asked questions (FAQs)

- **Q: How do I update Gemini CLI to the latest version?**
  - A: If you installed it globally via `npm`, update it using the command `npm install -g @google/gemini-cli@latest`. If you compiled it from source, pull the latest changes from the repository, and then rebuild using the command `npm run build`.

- **Q: Where are the Gemini CLI configuration or settings files stored?**
  - A: The Gemini CLI configuration is stored in two `settings.json` files:

    1. In your home directory: `~/.gemini/settings.json`.
    2. In your project's root directory: `./.gemini/settings.json`.
    
    Refer to [CLI Configuration](./cli/configuration.md) for more details.

- **Q: Why don't I see cached token counts in my stats output?**

>   - A: Currently, cached token information is only displayed when cached tokens are being used. This feature is currently available for API key users (Gemini API key or Google Cloud Vertex AI) but not for OAuth users (such as Google Personal/Enterprise accounts like Google Gmail or Google Workspace, respectively). This is because the Gemini Code Assist API does not support cached content creation. You can still view your total token usage with the `/stats` command.

Author: Maybe add "only, as in: "currently available for API key users only" ?

Reviewer: Can you drop the "currently" bits? Just to keep it "timeless". No idea where it is on the teams priority list.

  - **Solution:**
    Either stop the other process that is using the port or configure the MCP server to use a different port.

- **Error: Command not found (when attempting to run Gemini CLI with `gemini`).**
  - **Cause:** Gemini CLI is not correctly installed or it is not in your system's `PATH`.
  - **Solution:**
    The update depends on how you installed Gemini CLI:
    - If you installed `gemini` globally, check that your `npm` global binary directory is in your `PATH`. You can update Gemini CLI using the command `npm install -g @google/gemini-cli@latest`.
    - If you are running `gemini` from source, ensure you are using the correct command to invoke it (e.g., `node packages/cli/dist/index.js ...`). To update Gemini CLI, pull the latest changes from the repository, and then rebuild using the command `npm run build`.

- **Error: `MODULE_NOT_FOUND` or import errors.**
  - **Cause:** Dependencies are not installed correctly, or the project hasn't been built.
  - **Solution:**
    1.  Run `npm install` to ensure all dependencies are present.
    2.  Run `npm run build` to compile the project.

>     3.  Verify that the build completed successfully with `gemini --version`.

Reviewer: To avoid hitting  "not installed" or not on path, can you make this `npm run start`

Author: @markmcd PTAL
